From 8cdf7a2d2c0d766c46fec721b0c05c1a5984cc29 Mon Sep 17 00:00:00 2001
From: Steve Kondik <shade@chemlab.org>
Date: Fri, 6 Dec 2013 19:21:08 +0100
Subject: [PATCH 1/2] LatinIME: Add fallback loader for JNI lib

Prefer the JNI lib with gesture typing support.
If not available / installed fall back to the standard one without
geture typing.

Change-Id: I70a5f43386a9c72809a70bd286f3702378d6f9be
---
 .../src/com/android/inputmethod/latin/define/JniLibName.java   |  1 +
 java/src/com/android/inputmethod/latin/utils/JniUtils.java     | 10 +++++++---
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java b/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
index abfc36d..0fcf7ee 100644
--- a/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
+++ b/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
@@ -22,4 +22,5 @@ public final class JniLibName {
     }
 
     public static final String JNI_LIB_NAME = "jni_latinime";
+    public static final String JNI_LIB_NAME2 = "jni_unbundled_latinimegoogle";
 }
diff --git a/java/src/com/android/inputmethod/latin/utils/JniUtils.java b/java/src/com/android/inputmethod/latin/utils/JniUtils.java
index e7fdafa..0e4f582 100644
--- a/java/src/com/android/inputmethod/latin/utils/JniUtils.java
+++ b/java/src/com/android/inputmethod/latin/utils/JniUtils.java
@@ -25,9 +25,13 @@ public final class JniUtils {
 
     static {
         try {
-            System.loadLibrary(JniLibName.JNI_LIB_NAME);
-        } catch (UnsatisfiedLinkError ule) {
-            Log.e(TAG, "Could not load native library " + JniLibName.JNI_LIB_NAME, ule);
+            System.loadLibrary(JniLibName.JNI_LIB_NAME2);
+        } catch (UnsatisfiedLinkError ue) {
+            try {
+                System.loadLibrary(JniLibName.JNI_LIB_NAME);
+            } catch (UnsatisfiedLinkError ule) {
+                Log.e(TAG, "Could not load native library " + JniLibName.JNI_LIB_NAME, ule);
+            }
         }
     }
 
-- 
2.3.8 (Apple Git-58)


From 0b42920d1bddd18b88780a70482bd6854e0ea552 Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <ricardo@cyngn.com>
Date: Tue, 25 Nov 2014 11:55:27 +0000
Subject: [PATCH 2/2] Update JNI fallback lib name

Change-Id: I18db2d4cc87ffafc5a439a193f4d8f894136f5eb
---
 .../src/com/android/inputmethod/latin/define/JniLibName.java            | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java b/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
index 0fcf7ee..83ed5eb 100644
--- a/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
+++ b/java-overridable/src/com/android/inputmethod/latin/define/JniLibName.java
@@ -22,5 +22,5 @@ public final class JniLibName {
     }
 
     public static final String JNI_LIB_NAME = "jni_latinime";
-    public static final String JNI_LIB_NAME2 = "jni_unbundled_latinimegoogle";
+    public static final String JNI_LIB_NAME2 = "jni_latinimegoogle";
 }
-- 
2.3.8 (Apple Git-58)

